nixos/modules/services/web-apps/audiobookshelf.nix

```diff
@@ -0,0 +1,90 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.audiobookshelf;
+in
+{
+  options = {
+    services.audiobookshelf = {
+      enable = mkEnableOption "Audiobookshelf, self-hosted audiobook and podcast server.";
+
+      package = mkPackageOption pkgs "audiobookshelf" { };
+
+      dataDir = mkOption {
+        description = "Path to Audiobookshelf config and metadata inside of /var/lib.";
+        default = "audiobookshelf";
+        type = types.str;
+      };
+
+      host = mkOption {
+        description = "The host Audiobookshelf binds to.";
+        default = "127.0.0.1";
+        example = "0.0.0.0";
+        type = types.str;
+      };
+
+      port = mkOption {
+        description = "The TCP port Audiobookshelf will listen on.";
+        default = 8000;
+        type = types.port;
+      };
+
+      user = mkOption {
+        description = "User account under which Audiobookshelf runs.";
+        default = "audiobookshelf";
+        type = types.str;
+      };
+
+      group = mkOption {
+        description = "Group under which Audiobookshelf runs.";
+        default = "audiobookshelf";
+        type = types.str;
+      };
+
+      openFirewall = mkOption {
+        description = "Open ports in the firewall for the Audiobookshelf web interface.";
+        default = false;
+        type = types.bool;
+      };
```

Comment:
An alternative might be to configure an nginx reverse proxy. But it's ok not do that in this PR here.

Reply:
IMO since the reverse-proxy is so easy to set-up (one-line), there shouldn't be an option for it in the module.

Reply:
It would still be helpful for beginners. I wouldn't know this line by heart, and I'm not even a beginner anymore (I hope). Some [reverse proxy settings on github](https://github.com/advplyr/audiobookshelf#nginx-reverse-proxy) look sort of straightforward, but I'm still unsure whether you could fit them on a line readably.

Also a reverse proxy setup might include further options like a url.

Reply:
Beginners can read the documentation :-), `services.nginx.virtualHosts.<name>.locations."/".proxyPass = "http://127.0.0.1:<port>";` appears in multiple places.

We have multiple examples of setting-up Nginx for simple port forwarding.

There's no complex configuration to be done (websockets, sub-paths, PHP, auth, etc...), I don't think adding an option is necessary.

Reply:
See e.g. https://search.nixos.org/options?channel=23.05&show=services.xandikos.nginx.enable&from=0&size=50&sort=relevance&type=packages&query=nginx

Reply:
I see your point, but still I disagree. Anyways, we both agree that the present PR here is great without that option.